cpp/eval_node.cc

```diff
@@ -103,7 +106,7 @@ SumNode* SumNode::AddWordWork(
     }
   }
   if (!letter_child) {
-    letter_child = arena.NewSumNodeWithCapacity(num_choices == 1 ? 0 : 1);
+    letter_child = arena.NewSumNodeWithCapacity(__builtin_popcount(used_ordered) == 1 ? 0 : 1);
```

Comment:
Use `std::popcount` from `<bit>` (C++20)

Suggest pulling a meaningful name out here to improve legibility, e.g. `num_choices`.